hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -3417,6 +3417,17 @@ private void autoAdjustConfigsForConcurrencyMode(boolean isLockProviderPropertyS
             HoodieLockConfig.LOCK_PROVIDER_CLASS_NAME.key(), InProcessLockProvider.class.getName()));
       }
 
+      // for a single writer scenario, with all table services inline, lets set InProcessLockProvider
+      if (writeConfig.getWriteConcurrencyMode() == WriteConcurrencyMode.SINGLE_WRITER && !writeConfig.areAnyTableServicesAsync()) {
```

Comment:
Should this be the first check before `if (!isLockProviderPropertySet && writeConfig.isAutoAdjustLockConfigs() && isLockRequiredForSingleWriter())`?

Reply:
sure. fixed it.